### ISSUE TITLE
getRecentPerformanceSamples in web3.js breaks with custom commitment config

### ISSUE THREAD

**Issue by stegaBOB** [solana-labs/solana#26372]
The `getRecentPerformanceSamples` method in web3.js appends a commitment level to the POST request arguments if provided in the `Connection` object. RPC nodes currently do not expect this and return an error. This issue has been present for at least 1.5 months. I ran into this issue at the time but assumed it was an issue on my end and didn't dig further. 

#### Example code

```ts
(async () => {
  const connection1 = new Connection("https://ssc-dao.genesysgo.net");
  const connection2 = new Connection("https://ssc-dao.genesysgo.net", 'confirmed');

  try {
    const perf1 = await connection1.getRecentPerformanceSamples(1);
    console.log("Perf1 success: ", perf1[0]);
  } catch (e) {
    console.log("Perf1 failed: ", e);
  }

  try {
    const perf2 = await connection2.getRecentPerformanceSamples(1);
    console.log("Perf2 success: ", perf2[0]);
  } catch (e) {
    console.log("Perf2 failed: ", e);
  }
})();
```

#### Expected behavior: 
Both requests should return successfully, and the samples returned should be logged for both `perf1` and `perf2`. 

#### Current behavior:
`perf1` successfully logs and `perf2` throws an error as shown below:

<img width="581" alt="Screen Shot 2022-07-01 at 8 46 28 PM" src="https://user-images.githubusercontent.com/41593264/176982541-04e077b4-782a-4251-aa41-b293cc004ea4.png">

The payload of the `perf2` request: 
```json
{"method":"getRecentPerformanceSamples","jsonrpc":"2.0","params":[1,{"commitment":"confirmed"}],"id":"some random Id"}
```

The only difference is that `{"commitment":"confirmed"}` was added as an additional parameter. When I made an additional `Connection` with multiple options with a `ConnectionConfig`, the commitment level was still the only value added in the request.

#### Proposed Solution
None yet. Need to dig deeper into the web3.js code. This could either be fixed at a web3.js level, or the RPC client can be updated to accept the additional parameter.

**Comment by steveluscher** [comment]
Thank you for this! I’ll take a look.

**Comment by steveluscher** [comment]
Fixed in #26373!

**Comment by stegaBOB** [comment]
Wow that was fast! 👏
